include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -238,9 +241,16 @@ struct Base {
  */
 template <typename T>
 struct Unowned : T {
-  Unowned(typename T::contained_type* p) : T{p} {}
-  Unowned(Unowned&& v) : T{v.p_} {}
+  using Type = typename T::contained_type;
```

Comment:
Took these updates to C++ APIs from https://github.com/microsoft/onnxruntime/pull/12998

However, the above PR is actively changing, so I will integrate the final updates at a later date.